servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcUtils.java

```diff
@@ -347,7 +352,7 @@ private static CharSequence acceptedEncodingsHeaderValue0(final List<ContentCode
             builder.append(codec.name());
         }
 
-        return newAsciiString(builder.toString());
+        return builder.length() > 0 ? newAsciiString(builder.toString()) : null;
```

Comment:
suggestions:
- avoid conditional inside loop
- pre-size the string builder to reduce chances of copy/resize

```java
private static final String CONTENT_ENCODING_SEPARATOR = ", ";
```
```suggestion
        StringBuilder builder = new StringBuilder(codings.size() * (12 + CONTENT_ENCODING_SEPARATOR.length()));
        for (ContentCodec codec : codings) {
            if (codec == identity()) {
                continue;
            }

            builder.append(codec.name()).append(CONTENT_ENCODING_SEPARATOR);
        }
        if (builder.length() > CONTENT_ENCODING_SEPARATOR.length()) {
            builder.setLength(builder.length() - CONTENT_ENCODING_SEPARATOR.length());
            return newAsciiString(builder.toString());
        }
        return null;
```